packages/types/src/user/common.ts

```diff
@@ -1,5 +1,39 @@
-export type UserDTO = {
+import { DateComparisonOperator } from "../common"
+import { BaseFilterable } from "../dal"
+
+export interface UserDTO {
   id: string
+  email: string
+  first_name?: string | null
+  last_name?: string | null
+  avatar_url?: string | null
+  metadata?: Record<string, unknown> | null
+  created_at: Date
+  updated_at: Date
+  deleted_at?: Date | null
```

Comment:
suggestion: IMO, the DTO should represent the data corresponding to the User, properties should not be optional, if you retrieve the user all properties will be returned and eventually some will be null. Then comes the question of dynamic DTO's based on the select and relations but that is another topic :)

cc @olivermrbl what do you think?

Reply:
Good point, but I think this is in the same category as my other comment. I would leave it up to @pKorsholm to decide if we are tackling it now or later.